server/plugins/Slack.ts

```diff
@@ -64,29 +64,39 @@ export class Slack implements Notifier<Config> {
 	}
 
 	public async check(configTest: any): Promise<Config> {
+		const config = Slack.instanceOfConfig(configTest);
+
+		// If there no channels we don't have to make a request to check them
+		if (config.channels.length === 0) {
+			return config;
+		}
 
 		const qs = querystring.stringify({
 			token: this.token,
 			exclude_archived: true,
 			exclude_members: true
 		});
 
-		const config = Slack.instanceOfConfig(configTest);
-
-		const res = await fetch(`https://slack.com/api/channels.list?${qs}`);
-		const channels: { channels?: { name: string }[] } = await res.json();
-
-		const res2 = await fetch(`https://slack.com/api/groups.list?${qs}`);
-		const groups: { groups?: { name: string }[] } = await res2.json();
-
-		if (!groups.groups || !channels.channels) {
-			throw new Error("Could not verify slack channels.");
+		const destinations = await Promise.all([
+			fetch(`https://slack.com/api/channels.list?${qs}`)
+				.then(r => r.json())
+				.then((json: { channels?: { name: string }[] }) => {
+					return json.channels? json.channels.map(c => c.name) : json;
+				}),
+			fetch(`https://slack.com/api/groups.list?${qs}`)
+				.then(r => r.json())
+				.then((json: { groups?: { name: string }[] }) => {
+					return json.groups? json.groups.map(c => c.name) : json;
+				})
+		]);
+
+		for (const dest of destinations) {
+			if (!Array.isArray(dest)) {
```

Comment:
Maybe you should check that the elements in here are expected type of names, i.e. strings? (If false, it'd get caught below, so still safe, but a suggestion)

Reply:
They don't need to be strings, because a `null === "somestring"` will still be false.